Public/Src/Engine/Dll/Distribution/Grpc/ClientConnectionManager.cs

```diff
@@ -69,20 +69,43 @@ public async Task MonitorConnectionAsync()
 
                 if (state == ChannelState.Idle)
                 {
-                    // Try connecting with timeout (default 5min) 
-                    // If it does not succeed, disconnect the worker. 
-                    bool connectionSucceeded = await TryConnectChannelAsync(GrpcSettings.CallTimeout, nameof(MonitorConnectionAsync));
-                    if (!connectionSucceeded && IsNonRecoverableState(Channel.State))
+                    bool isReconnected = await TryReconnect();
+                    if (!isReconnected)
                     {
-                        // If re-connection attempt does not succeed; but the state is one of the recoverable states (Connecting, Ready, Transient)
-                        // Then, there is still a chance for the reconnection. 
                         OnConnectionTimeOutAsync?.Invoke(this, EventArgs.Empty);
                         break;
                     }
                 }
             }
         }
 
+        private async Task<bool> TryReconnect()
+        {
+            int numRetries = 0;
+            bool connectionSucceeded = false;
+
+            while (numRetries < GrpcSettings.MaxRetry)
+            {
+                numRetries++;
+
+                // Try connecting with timeout (default 5min) 

```

Comment:
>(default 5min) [](start = 47, length = 14)

nit: I wouldn't put this here; if we change the defaults later, we'll need it to update the comment as well...